r/tests/testthat/test-dplyr-string-functions.R

```diff
@@ -868,42 +868,76 @@ test_that("str_like", {
 })
 
 test_that("str_pad", {
-  
+
   df <- tibble(x = c("Foo and bar", "baz and qux and quux"))
-  
+
   expect_dplyr_equal(
     input %>%
       mutate(x = str_pad(x, width = 31)) %>%
       collect(),
     df
   )
-  
+
   expect_dplyr_equal(
     input %>%
       mutate(x = str_pad(x, width = 30, side = "right")) %>%
       collect(),
     df
   )
-  
+
   expect_dplyr_equal(
     input %>%
       mutate(x = str_pad(x, width = 31, side = "left", pad = "+")) %>%
       collect(),
     df
   )
-  
+
   expect_dplyr_equal(
     input %>%
       mutate(x = str_pad(x, width = 10, side = "left", pad = "+")) %>%
       collect(),
     df
   )
-  
+
   expect_dplyr_equal(
     input %>%
       mutate(x = str_pad(x, width = 31, side = "both")) %>%
       collect(),
     df
   )
-  
+})
+
+test_that("substr, substring, str_sub", {
+
+  df <- tibble(x1 = "Apache Arrow")
+
+  expect_dplyr_equal(
+    input %>%
+      mutate(
+        x2 = substr(x1, 1, 6), # Apache
+        x3 = substr(x1, 8, 12) # Arrow
+      ) %>%
+      collect(),
+    df
+  )
+
+  expect_dplyr_equal(
+    input %>%
+      mutate(
+        x2 = substring(x1, 1, 6), # Apache
+        x3 = substring(x1, 8, 12) # Arrow
+      ) %>%
+      collect(),
+    df
+  )
+
+  expect_dplyr_equal(
+    input %>%
+      mutate(
+        x2 = str_sub(x1, 1, 6), # Apache
+        x3 = str_sub(x1, 8, 12) # Arrow
+      ) %>%
+      collect(),
+    df
+  )
 })
```

Comment:
The tests are now way shorter than they were before and don't include any of those awkward test cases like negative numbers that we were discussing on the ticket - maybe we should add them back in?

Reply:
yes !! i have no idea what happened here, but i'll re-add or re-write those